core/src/parking_lot.rs

```diff
@@ -28,6 +31,15 @@ use smallvec::SmallVec;
 // around the stdio file descriptors.
 // Here we use #[linkage] and #[export_name] to make both versions point to
 // the same instance.
+//
+// REVIEW: Hm yeah something else will be needed here, we shouldn't be dealing
+// with mangling like this and/or symbol linkage, it's just asking for problems.
+// It's long been thought that unit tests of libstd itself are just way too
+// difficult to deal with and this could be the nail in the coffin. If this
+// can't be fixed by other means then all unit tests in libstd should be
+// extracted to standalone tests in `src/libstd/tests` in the same manner as
+// `src/libcore/tests` and then unit tests for the libstd library should be
+// disabled.
```

Comment:
I'm not deeply familiar with what tests are present in `libstd` currently. But moving them out to a separate `tests/` makes them only able to access the public API of `libstd`, right? So any test that tests some internal function or other aspect will not work this way.

Simply not allowing testing anything internal in the entire standard library seems like a huge drawback to me, but as I said, I'm not familiar with the current tests and the approach they are usually written with.

Reply:
That is correct, yes, we can't easily test private internal implementation details if they're moved out. That being said this is already the case for libcore, and there's really not that many private implementation details of libstd which aren't public in one way or another. Currently libcore gets around this by having unstable exports to test.

Like I'm sorry but a mangled constant with weird linkage is just a hard no for putting this in the standard library. This is basically guaranteed to cause us unending pain through LLVM updates, tiny linkage tweaks etc. This issue needs to be put to rest at the beginning one way or another which doesn't involve dipping so far into unstable features.